templates/al2023/runtime/rootfs/etc/systemd/system/nodeadm-run.service

```
@@ -8,7 +8,8 @@ Requires=nodeadm-config.service
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/nodeadm init --skip config
+# ignores IMDS as a config source, because nodeadm-config should have cached the config.
+ExecStart=/usr/bin/nodeadm init --skip config --config-source file:///etc/eks/nodeadm.d/ --config-cache /run/eks/nodeadm/config.json
```

this is still required to contain imds source because we want the full NodeConfig spec including the defaults from config that are the userdata part.
```suggestion
ExecStart=/usr/bin/nodeadm init --skip config --config-source imds://user-data,file:///etc/eks/nodeadm.d/ --config-cache /run/eks/nodeadm/config.json
```